sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -488,16 +477,13 @@ class Analyzer(override val catalogManager: CatalogManager) extends RuleExecutor
             case e if !e.resolved => u
             case g: Generator => MultiAlias(g, Nil)
             case c @ Cast(ne: NamedExpression, _, _, _) => Alias(c, ne.name)()
-            case e: ExtractValue =>
-              if (extractOnly(e)) {
-                Alias(e, toPrettySQL(e))()
-              } else {
-                Alias(e, toPrettySQL(e))(explicitMetadata = Some(metaForAutoGeneratedAlias))
-              }
             case e if optGenAliasFunc.isDefined =>
               Alias(child, optGenAliasFunc.get.apply(e))()
             case l: Literal => Alias(l, toPrettySQL(l))()
             case e =>
+              val metaForAutoGeneratedAlias = new MetadataBuilder()
+                .putString("__autoGeneratedAlias", "true")
+                .build()
```

Comment:
To be safe and not change DataFrame behavior, can we not add this metadata if the expression is extractOnly?

Reply:
done. Please, double check.